### PR TITLE
DR: Remove components proxying requests to Kubernetes API server

### PR DESCRIPTION
**Pull request by pbochynski**
Update decision records for https://github.com/kyma-project/community/issues/576

### CODE REVIEW

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+In consequence there is no pre-configured auth model for Grafana/Kiali/Jaeger exposure. We decided to have an oauth2-proxy in front of all this UIs having a default configuration which points to a documentation. Here port-forwarding is described as well as a guide on how to configure the proxy with OIDC, see also https://github.com/kyma-project/kyma/issues/11181
+
+The benefits:
+- faster installation and lower resource consumption (6 components less),
+- reduced complexity of authentication and authorization flow
+- improved security (e.g. removed service accounts with powerful roles)
+- faster development and easier maintenance of Console UI 
+
+As a result of this decision following decisions are invalidated:
```
Comment (mmitoraj) [inline collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md:44]:
```suggestion
As a result of this decision, the following decisions are invalidated:
```

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+- [DR 007: GraphQL as API facade for UI](https://github.com/kyma-project/community/blob/main/collaboration/sig-core/decisions/dr-007-GraphQL_as_API_facade_for_UI.md)
+- [DR 015: Authorization for GraphQL](https://github.com/kyma-project/community/blob/main/collaboration/sig-core/decisions/dr-015-Authorization_for_GraphQL.md)
```
Comment (mmitoraj) [inline collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md:46]: If you want to list them chronologically then **DR 008** should go before **DR 015**.

**Review by mmitoraj — CHANGES_REQUESTED** [review]
Hi, please see my comments below. Thank you :)

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Decision log
+
+| Name | Description |
+|-----------------------|------------------------------------------------------------------------------------|
+| Title | Remove components proxying requests to Kubernetes API server |
+| Due date | 2021-07-19 |
+| Status | Proposed on 2021-07-12, Accepted on 2021-08-04|
+| Decision type | Binary |
+| Affected decisions | [DR 007: GraphQL as API facade for UI](https://github.com/kyma-project/community/blob/main/collaboration/sig-core/decisions/dr-007-GraphQL_as_API_facade_for_UI.md), [DR 015: Authorization for GraphQL](https://github.com/kyma-project/community/blob/main/collaboration/sig-core/decisions/dr-015-Authorization_for_GraphQL.md), [DR 008: DEX as an OIDC authenticator](https://github.com/kyma-project/community/blob/main/collaboration/sig-core/decisions/dr-008-Dex_as_an_OIDC_authenticator.md) |
+
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md:14]: I think we had such a verbose template earlier. Now it is simpler and it is part of the issue template: https://github.com/kyma-project/community/blob/main/.github/ISSUE_TEMPLATE/decision-record.md

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md (hunk 2 above)
Comment (pbochynski) [reply collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md:46]: I wanted to put related decisions (GraphQL) together. Chronological order is not relevant here.

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+In consequence, there is no pre-configured Auth model for Grafana, Kiali, or Jaeger exposure. We decided to have an [OAuth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy) in front of all these UIs with a default configuration that points to documentation. Here port-forwarding is described as well as a guide on how to configure the proxy with OIDC, see also https://github.com/kyma-project/kyma/issues/11181
+
+The benefits:
+- faster installation and lower resource consumption (6 components removed),
+- reduced complexity of the authentication and authorization flow,
+- improved security (e.g. removed service accounts with powerful roles),
+- faster development and easier maintenance of the Console UI.
+
+As a result of this decision, the following decisions are invalidated:
```
Comment (mmitoraj) [inline collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md:44]:
```suggestion
As a result of this decision, the following decisions are invalid:
```

**Review by mmitoraj — COMMENTED** [review]
- Please let me know what you meant in line 36? I re-posted my comment and added a few minor ones that came to my mind when I was reading the doc again.  
- Re: `collaboration/sig-core/decisions/dr-008-Dex_as_an_OIDC_authenticator.md` and `collaboration/sig-core/decisions/dr-015-Authorization_for_GraphQL.md` shouldn't we update the **Status** section and change it to "Invalidated on 2021-08-04."?
Thank you.

File: collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- dex
+
+To replace the removed components, use the Kubernetes API server and its features for authentication and authorization, i.e.:
+- [Authentication with OpenId Connect Tokens](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#openid-connect-tokens)
+- [Role-based access control (RBAC) Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+
+
+## Consequences
+
+In consequence, there is no pre-configured Auth model for Grafana, Kiali, or Jaeger exposure. We decided to have an [OAuth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy) in front of all these UIs with a default configuration that points to documentation. Here port-forwarding is described as well as a guide on how to configure the proxy with OIDC, see also https://github.com/kyma-project/kyma/issues/11181
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply collaboration/sig-core/decisions/dr-021-Remove-k8s-api-server-proxies.md]: I removed that sentence. It is not needed here. We have already a link between https://github.com/kyma-project/community/issues/576 and https://github.com/kyma-project/kyma/issues/11181

**Comment by pbochynski** [comment]
I applied review comments.

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**